### PR TITLE
57: Hide credentials passed in the URL in the load test report

### PR DESCRIPTION
**Pull request by h-arlt**
- introduce new property to control whether or not to remove user-info from request URL
- remove user-info from request URL if need to do so before writing timer data to CSV
- added tests for new method 'removeUserInfo' of utility class 'com.xceptance.xlt.engine.util.UrlUtils'
- added documentation of new property to all 'default.properties' files in sample test suites

### CODE REVIEW

File: src/main/java/com/xceptance/xlt/engine/util/UrlUtils.java [hunk 1]
```diff
@@ -223,4 +225,37 @@ public static String getUrlEncodedParameters(final List<NameValuePair> parameter
...
+            }
+            if (info.getFragment() != null)
+            {
+                sb.append('#').append(info.getFragment());
+            }
+            return sb.toString();
+        }
+
+        return null;
+    }
```
Comment (jowerner) [inline src/main/java/com/xceptance/xlt/engine/util/UrlUtils.java:260]: Small performance optimization: We often have URL objects, not strings. What about providing a second version of `UrlUtils.removeUserInfo()` that takes a URL and builds the resulting user-less url string from the already parsed components in URL? This way we don't need to convert the URL to a string and afterwards parse the string again into its pieces.

File: src/main/java/com/xceptance/xlt/engine/XltHttpWebConnection.java [hunk 2]
```diff
@@ -486,4 +486,17 @@ protected static void putAdditionalRequestData(RequestData requestData, WebReque
...
+
+        // remove user-info from request URL if we need to (GH #57)
+        if(SessionImpl.REMOVE_USERINFO_FROM_REQUEST_URL)
+        {
+            urlString = UrlUtils.removeUserInfo(urlString);
+        }
+
+        return urlString;
+    }
 }
```
Comment (jowerner) [inline src/main/java/com/xceptance/xlt/engine/XltHttpWebConnection.java:502]: In the result browser, the user info is still present in the URLs. Maybe we can fix this by removing the user info in `XltHttpWebConnection.cloneWebRequest()`?

Comment (h-arlt) [reply src/main/java/com/xceptance/xlt/engine/XltHttpWebConnection.java:520]: Done.

However, I'm not absolutely sure that this is a good idea as the result browser is primarily used for debugging and analysis purposes and users might wonder why this information is removed from there as well. As for me, I'd expect the result browser to represent the data that actually transmitted.

File: src/main/java/com/xceptance/xlt/engine/util/UrlUtils.java (hunk 1 above)
Comment (h-arlt) [reply src/main/java/com/xceptance/xlt/engine/util/UrlUtils.java:260]: Done.

File: src/main/java/com/xceptance/xlt/engine/XltHttpWebConnection.java (hunk 2 above)
Comment (jowerner) [reply src/main/java/com/xceptance/xlt/engine/XltHttpWebConnection.java:527]: Result browsers are often linked to in load test reports. If these reports are shared carelessly with other people, they might see this info again. So better safe than sorry.

**Review by jowerner — APPROVED** [review]

**Pull request merged**